[CI] ReindexNodeShutdownIT testReindexWithShutdown failing

**Build Scans:**
- [elasticsearch-intake #33076 / part1](https://gradle-enterprise.elastic.co/s/tc3nxl3q57rfu)
- [elasticsearch-pull-request #113262 / part-1](https://gradle-enterprise.elastic.co/s/xtvqrqo2of3ts)
- [elasticsearch-pull-request #113251 / part-1](https://gradle-enterprise.elastic.co/s/pxijtcy62qndy)
- [elasticsearch-periodic-platform-support #11674 / ubuntu-2404-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/myir7qzunpv5o)
- [elasticsearch-pull-request #112778 / part-1](https://gradle-enterprise.elastic.co/s/wlyrsh6iaspva)
- [elasticsearch-periodic-platform-support #11669 / ubuntu-2404-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/u6al3ctv2qi22)
- [elasticsearch-periodic #11696 / openjdk25_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/3t33x23t5keky)
- [elasticsearch-intake #32947 / part1](https://gradle-enterprise.elastic.co/s/7vdx7gbfw5fcy)

**Reproduction Line:**
```
./gradlew ":modules:reindex:internalClusterTest" --tests "org.elasticsearch.index.reindex.ReindexNodeShutdownIT.testReindexWithShutdown" -Dtests.seed=D41662F44CD83375 -Dtests.locale=ksf-CM -Dtests.timezone=America/North_Dakota/Center -Druntime.java=25
```

**Applicable branches:**
9.2

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.reindex.ReindexNodeShutdownIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testReindexWithShutdown'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Encounterd Failed to execute phase [query], all shards failed; shardFailures {[pGFZ2JKQSR2PLdwmLWmIQA][reindex-shutdown-index][0]: org.elasticsearch.transport.NodeDisconnectedException: [node_t1][127.0.0.1:31512][indices:data/read/search[query][n]] disconnected
}{[pGFZ2JKQSR2PLdwmLWmIQA][reindex-shutdown-index][1]: org.elasticsearch.transport.NodeDisconnectedException: [node_t1][127.0.0.1:31512][indices:data/read/search[query][n]] disconnected
}{[pGFZ2JKQSR2PLdwmLWmIQA][reindex-shutdown-index][2]: org.elasticsearch.transport.NodeDisconnectedException: [node_t1][127.0.0.1:31512][indices:data/read/search[query][n]] disconnected
}{[pGFZ2JKQSR2PLdwmLWmIQA][reindex-shutdown-index][3]: org.elasticsearch.transport.NodeDisconnectedException: [node_t1][127.0.0.1:31512][indices:data/read/search[query][n]] disconnected
}{[pGFZ2JKQSR2PLdwmLWmIQA][reindex-shutdown-index][4]: org.elasticsearch.transport.NodeDisconnectedException: [node_t1][127.0.0.1:31512][indices:data/read
[truncated]
```

**Issue Reasons:**
- [9.2] 2 consecutive failures in step ubuntu-2404-aarch64_checkpart1_platform-support-arm
- [9.2] 8 failures in test testReindexWithShutdown (25.0% fail rate in 32 executions)
- [9.2] 2 failures in step part1 (28.6% fail rate in 7 executions)
- [9.2] 3 failures in step part-1 (33.3% fail rate in 9 executions)
- [9.2] 2 failures in step ubuntu-2404-aarch64_checkpart1_platform-support-arm (100.0% fail rate in 2 executions)
- [9.2] 2 failures in pipeline elasticsearch-intake (28.6% fail rate in 7 executions)
- [9.2] 3 failures in pipeline elasticsearch-pull-request (33.3% fail rate in 9 executions)
- [9.2] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/ReindexNodeShutdownIT.java
+++ b/modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/ReindexNodeShutdownIT.java
@@ -103,8 +103,8 @@ public class ReindexNodeShutdownIT extends ESIntegTestCase {
 
             @Override
             public void onFailure(Exception e) {
-                logger.debug("Encounterd " + e.toString());
-                fail(e, "Encounterd " + e.toString());
+                logger.debug("Encountered " + e.toString());
+                fail(e, "Encountered " + e.toString());
             }
         };
         internalCluster().client(coordNodeName).execute(ReindexAction.INSTANCE, reindexRequest, reindexListener);

--- a/server/src/main/java/org/elasticsearch/node/NodeConstruction.java
+++ b/server/src/main/java/org/elasticsearch/node/NodeConstruction.java
@@ -1241,7 +1241,7 @@ class NodeConstruction {
             onlinePrewarmingService
         );
 
-        final var shutdownPrepareService = new ShutdownPrepareService(settings, httpServerTransport, taskManager, terminationHandler);
+        final var shutdownPrepareService = new ShutdownPrepareService(settings, httpServerTransport, transportService, terminationHandler);
 
         modules.add(loadPersistentTasksService(settingsModule, clusterService, threadPool, clusterModule.getIndexNameExpressionResolver()));
 

--- a/server/src/main/java/org/elasticsearch/node/ShutdownPrepareService.java
+++ b/server/src/main/java/org/elasticsearch/node/ShutdownPrepareService.java
@@ -22,6 +22,7 @@ import org.elasticsearch.http.HttpServerTransport;
 import org.elasticsearch.index.reindex.ReindexAction;
 import org.elasticsearch.node.internal.TerminationHandler;
 import org.elasticsearch.tasks.TaskManager;
+import org.elasticsearch.transport.TransportService;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -68,7 +69,7 @@ public class ShutdownPrepareService {
     public ShutdownPrepareService(
         Settings settings,
         HttpServerTransport httpServerTransport,
-        TaskManager taskManager,
+        TransportService transportService,
         TerminationHandler terminationHandler
     ) {
         this.maxTimeout = MAXIMUM_SHUTDOWN_TIMEOUT_SETTING.get(settings);
@@ -76,8 +77,8 @@ public class ShutdownPrepareService {
 
         final var reindexTimeout = MAXIMUM_REINDEXING_TIMEOUT_SETTING.get(settings);
         addShutdownHook("http-server-transport-stop", httpServerTransport::close);
-        addShutdownHook("async-search-stop", () -> awaitSearchTasksComplete(maxTimeout, taskManager));
-        addShutdownHook("reindex-stop", () -> awaitReindexTasksComplete(reindexTimeout, taskManager));
+        addShutdownHook("async-search-stop", () -> awaitSearchTasksComplete(maxTimeout, transportService.getTaskManager()));
+        addShutdownHook("reindex-stop", () -> awaitReindexTasksComplete(reindexTimeout, transportService.getTaskManager()));
         if (terminationHandler != null) {
             addShutdownHook("termination-handler-stop", terminationHandler::handleTermination);
         }

